src/postgres/src/backend/utils/misc/pg_yb_utils.c

```diff
@@ -3575,30 +3575,98 @@ void YbSetIsBatchedExecution(bool value)
 	yb_is_batched_execution = value;
 }
 
-void ProcSetNodeUUID(const char *node_uuid) {
-	strcpy(MyProc->node_uuid, node_uuid);
+void ProcSetTopLevelNodeId(const uint64_t *top_level_node_id) {
+	MyProc->top_level_node_id[0] = top_level_node_id[0];
+	MyProc->top_level_node_id[1] = top_level_node_id[1];
 }
 
 void ProcSetTopRequestId(const uint64_t *top_level_request_id) {
 	MyProc->top_level_request_id[0] = top_level_request_id[0];
 	MyProc->top_level_request_id[1] = top_level_request_id[1];
 }
 
-void
-top_level_request_id_uint_to_char(char *top_level_request_id, const uint64_t top_level_request_id_uint[2])
-{
-    uint64_t nth_request_id = top_level_request_id_uint[0];
-    int index = 15;
-    for (; index >= 0; index--)
-    {
-      if (index == 8)
-        nth_request_id = top_level_request_id_uint[1];
-      if (nth_request_id % 16 < 10)
-        top_level_request_id[index] = '0' + (nth_request_id % 16);
-      else
-        top_level_request_id[index] = 'a' + ((nth_request_id % 16) % 10);
-      nth_request_id /= 10;
-    }
+void uint128_to_char(char *char_id, const uint64_t uint_id[2])
+{
+	sprintf(char_id, "%llx%llx", uint_id[0], uint_id[1]);
+}
+
+uint32 remote_host_port_to_uint(const char* remote_host)
+{
+	uint32 client_node_ip = 0;
+	int bit_position = 0, index;
+	for (index = strlen(remote_host) - 1; index >= 0; index--, bit_position += 8)
+	{
+		int power = 1, octate = 0, bit;
+		while (index >= 0 && remote_host[index] != '.')
+		{
+			octate += power * (remote_host[index--] - '0');
+			power *= 10;
+		}
+		for (bit = 0; bit < 8; bit++)
+		{
+			if ((octate >> bit) & 1)
+			{
+				client_node_ip ^= (1 << (bit_position + bit));
+			}
+		}
+	}
+	return client_node_ip;
+}
+
+int num_digits(uint16 digit)
+{
+	int digits = 0;
+	while (digit > 0)
+	{
+		digits++;
+		digit /= 10;
+	}
+	return digits;
+}
+
+void client_node_ip_to_string(char *client_node_ip,
```

Comment:
Generally as per our coding guidelines, we'd want to have the input parameters first, then followed by output parameters.

So unless postgres requires a different convention, the output `char *client_node_ip` should become the 3rd parameter.